LRM/settings.md
### **LRM/settings.py**
```python
"""
Django settings for LRM project.

Generated by 'django-admin startproject' using Django 4.1.4.

For more information on this file, see
https://docs.djangoproject.com/en/4.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.1/ref/settings/
"""

from pathlib import Path
import os
import socket

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent

def get_ip_address():
    s=socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    s.connect(("8.8.8.8", 80))
    ip = s.getsockname()[0]
    return ip

SERVER_IP= get_ip_address()
# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-y8ijc!o3a25ki6-@)x+*q44%lo#vpi$s3$1wv794#z-nd=vvr^'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['*']

# Application definition

INSTALLED_APPS = [
    'simpleui',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'admin_model_list_order',
    'lrm',
    'pure_pagination',
    'bootstrap3',
    'widget_tweaks', # 项目依赖
    'django_tables2',# 项目依赖
    'django_filters',# 项目依赖
    'import_export'
]

#IMPORT_EXPORT_USE_TRANSACTIONS = True

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'admin_model_list_order.middleware.AdminModelListOrder',
]

ADMIN_MODEL_DEFAULT_PRIORITY = 100

ROOT_URLCONF = 'LRM.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates'),'templates'],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'LRM.wsgi.application'

# Database
# https://docs.djangoproject.com/en/4.1/ref/settings/#databases

DATABASES = {
    'default': {
        #'ENGINE': 'django.db.backends.sqlite3',
        #'NAME': BASE_DIR / 'db.sqlite3',
        'ENGINE': 'django.db.backends.mysql',
        'HOST': '10.57.170.40',
        'PORT': '3306',
        'NAME': 'lrm_database',
        'USER': 'dytopo',
        'PASSWORD': 'Ossdbg1*',
    }
}

# Password validation
# https://docs.djangoproject.com/en/4.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Internationalization
# https://docs.djangoproject.com/en/4.1/topics/i18n/

LANGUAGE_CODE = 'zh-hans'
TIME_ZONE = 'Asia/Shanghai'

USE_I18N = True
USE_TZ = False

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.1/howto/static-files/

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media') # 默认根目录

STATIC_URL = '/static/'

STATIC_ROOT = os.path.join(BASE_DIR, "static")

# Default primary key field type
# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
SESSION_ENGINE = 'django.contrib.sessions.backends.cache'

LOGIN_URL = '/admin/users/login'
SIMPLEUI_HOME_INFO = False 
SIMPLEUI_ANALYSIS = False 
# 修改左侧菜单首页设置
SIMPLEUI_HOME_PAGE = 'http://%s:8000/' % SERVER_IP # 指向页面
SIMPLEUI_HOME_TITLE = '自动化中台主页' # 首页标题
SIMPLEUI_HOME_ICON = 'fas fa-home' # 首页图标
SIMPLEUI_ICON = {
    '设备资源': 'fa-sharp fa-solid fa-server',
    '设备端口资源': 'fa-sharp fa-solid fa-ethernet',
    '交换机端口资源': 'fa-sharp fa-solid fa-network-wired',
    '端口连接信息': 'fa-sharp fa-solid fa-exchange',
    '思博伦虚拟测试仪认证服务器信息': 'fa-sharp fa-solid fa-key',
}
DATA_UPLOAD_MAX_NUMBER_FIELDS = None

SIMPLEUI_CONFIG = {
    'system_keep': True,
    'dynamic': True,  # 设置是否开启动态菜单, 默认为False. 如果开启, 则会在每次用户登陆时动态展示菜单内容
    'menus': [
        {
            'name': '数据可视化',
            'icon': 'fa-fw fa fa-area-chart',
            'models': [{
                'name': '绘制设备组网拓扑图',
                'icon': 'fa-fw fa fa-pencil',
                'url': 'http://%s:8088/' % SERVER_IP
            }]
        },
    ]
}

```


LRM/settings.py
"""
Django settings for LRM project.

Generated by 'django-admin startproject' using Django 4.1.4.

For more information on this file, see
https://docs.djangoproject.com/en/4.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.1/ref/settings/
"""

from pathlib import Path
import os
import socket

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent

def get_ip_address():
    s=socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    s.connect(("8.8.8.8", 80))
    ip = s.getsockname()[0]
    return ip

SERVER_IP= get_ip_address()
# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-y8ijc!o3a25ki6-@)x+*q44%lo#vpi$s3$1wv794#z-nd=vvr^'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['*']

# Application definition

INSTALLED_APPS = [
    'simpleui',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'admin_model_list_order',
    'lrm',
    'pure_pagination',
    'bootstrap3',
    'widget_tweaks', # 项目依赖
    'django_tables2',# 项目依赖
    'django_filters',# 项目依赖
    'import_export'
]

#IMPORT_EXPORT_USE_TRANSACTIONS = True

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'admin_model_list_order.middleware.AdminModelListOrder',
]

ADMIN_MODEL_DEFAULT_PRIORITY = 100

ROOT_URLCONF = 'LRM.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates'),'templates'],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'LRM.wsgi.application'

# Database
# https://docs.djangoproject.com/en/4.1/ref/settings/#databases

DATABASES = {
    'default': {
        #'ENGINE': 'django.db.backends.sqlite3',
        #'NAME': BASE_DIR / 'db.sqlite3',
        'ENGINE': 'django.db.backends.mysql',
        'HOST': '10.57.170.40',
        'PORT': '3306',
        'NAME': 'lrm_database',
        'USER': 'dytopo',
        'PASSWORD': 'Ossdbg1*',
    }
}

# Password validation
# https://docs.djangoproject.com/en/4.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

# Internationalization
# https://docs.djangoproject.com/en/4.1/topics/i18n/

LANGUAGE_CODE = 'zh-hans'
TIME_ZONE = 'Asia/Shanghai'

USE_I18N = True
USE_TZ = False

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.1/howto/static-files/

MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media') # 默认根目录

STATIC_URL = '/static/'

STATIC_ROOT = os.path.join(BASE_DIR, "static")

# Default primary key field type
# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
SESSION_ENGINE = 'django.contrib.sessions.backends.cache'

LOGIN_URL = '/admin/users/login'
SIMPLEUI_HOME_INFO = False 
SIMPLEUI_ANALYSIS = False 
# 修改左侧菜单首页设置
SIMPLEUI_HOME_PAGE = 'http://%s:8000/' % SERVER_IP # 指向页面
SIMPLEUI_HOME_TITLE = '自动化中台主页' # 首页标题
SIMPLEUI_HOME_ICON = 'fas fa-home' # 首页图标
SIMPLEUI_ICON = {
    '设备资源': 'fa-sharp fa-solid fa-server',
    '设备端口资源': 'fa-sharp fa-solid fa-ethernet',
    '交换机端口资源': 'fa-sharp fa-solid fa-network-wired',
    '端口连接信息': 'fa-sharp fa-solid fa-exchange',
    '思博伦虚拟测试仪认证服务器信息': 'fa-sharp fa-solid fa-key',
}
DATA_UPLOAD_MAX_NUMBER_FIELDS = None

SIMPLEUI_CONFIG = {
    'system_keep': True,
    'dynamic': True,  # 设置是否开启动态菜单, 默认为False. 如果开启, 则会在每次用户登陆时动态展示菜单内容
    'menus': [
        {
            'name': '数据可视化',
            'icon': 'fa-fw fa fa-area-chart',
            'models': [{
                'name': '绘制设备组网拓扑图',
                'icon': 'fa-fw fa fa-pencil',
                'url': 'http://%s:8088/' % SERVER_IP
            }]
        },
    ]
}
